pkg/networkservice/common/interpose/server_test.go

```diff
@@ -14,131 +14,109 @@
+		adapters.NewServerToClient(next.NewNetworkServiceServer(
+			// actual `connect.NewServer()` should not update `request.Connection.Path.PathIndex`
+			new(restorePathServer),
+			updatepath.NewServer("nsmgr"),
+			clienturl.NewServer(&nseURL),
+			interposeServer,
+			checkcontext.NewServer(t, func(t *testing.T, ctx context.Context) {
+				clientURL := clienturlctx.ClientURL(ctx)
+				require.NotNil(t, clientURL)
+				require.Equal(t, crossNSEURL, *clientURL)
+			}),
+		)),
+		adapters.NewServerToClient(next.NewNetworkServiceServer(
+			new(restorePathServer),
+			updatepath.NewServer("interpose-nse"),
+		)),
+		adapters.NewServerToClient(next.NewNetworkServiceServer(
+			new(restorePathServer),
+			updatepath.NewServer("nsmgr"),
+			clienturl.NewServer(&nseURL),
+			interposeServer,
+			checkcontext.NewServer(t, func(t *testing.T, ctx context.Context) {
+				clientURL := clienturlctx.ClientURL(ctx)
+				require.NotNil(t, clientURL)
+				require.Equal(t, nseURL, *clientURL)
+			}),
+		)),
+	)
 
-type copyClient struct {
-	requests []*networkservice.NetworkServiceRequest
-}
+	// 1. Request
+
+	request := &networkservice.NetworkServiceRequest{
+		Connection: &networkservice.Connection{},
+	}
+
+	conn, err := client.Request(context.TODO(), request)
+	require.NoError(t, err)
+
+	// 2. Refresh
 
-func (c *copyClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
-	request = request.Clone()
-	c.requests = append(c.requests, request)
-	// To be sure it is not modified after and we could check contents.
 	request = request.Clone()
-	return next.Client(ctx).Request(ctx, request)
-}
+	request.Connection = conn.Clone()
 
-func (c *copyClient) Close(ctx context.Context, conn *networkservice.Connection, opts ...grpc.CallOption) (*empty.Empty, error) {
-	conn = conn.Clone()
-	return next.Client(ctx).Close(ctx, conn)
-}
+	conn, err = client.Request(context.TODO(), request)
+	require.NoError(t, err)
 
-func TestCrossNSERequest(t *testing.T) {
-	var regServer registry.NetworkServiceEndpointRegistryServer
-	crossURL := "test://crossconnection"
-	clientURL := &url.URL{
-		Scheme: "unix",
-		Path:   "/var/run/nse-1.sock",
-	}
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	server := endpoint.NewServer(ctx, "nsmgr",
-		authorize.NewServer(),
-		TokenGenerator,
-		interpose.NewServer("nsmgr", &regServer))
-
-	regClient := next_reg.NewNetworkServiceEndpointRegistryClient(interpose_reg.NewNetworkServiceEndpointRegistryClient(), adapters2.NetworkServiceEndpointServerToClient(regServer))
-
-	reg, err := regClient.Register(context.Background(), &registry.NetworkServiceEndpoint{
-		Name: "cross-nse",
-		Url:  crossURL,
-	})
-	require.Nil(t, err)
-	require.NotNil(t, reg)
+	// 3. Close
 
-	interposeRegName := reg.Name
-	interposeNSE := endpoint.NewServer(ctx, interposeRegName,
-		authorize.NewServer(),
-		TokenGenerator)
+	conn = conn.Clone()
```

Comment:
Can we add a chain element into the end to check that Close has been called on the last chain element(NSE)

Reply:
Added `touchServer` to verify that every `Request`, `Close` is invoked on the last chain element.